shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/IstioSetupService.java

```diff
@@ -0,0 +1,121 @@
+package com.redhat.service.smartevents.shard.operator;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.smartevents.shard.operator.providers.TemplateProvider;
+import com.redhat.service.smartevents.shard.operator.resources.istio.gateway.Gateway;
+import com.redhat.service.smartevents.shard.operator.resources.istio.requestauthentication.RequestAuthentication;
+import com.redhat.service.smartevents.shard.operator.resources.istio.virtualservice.VirtualService;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.quarkus.runtime.StartupEvent;
+
+@ApplicationScoped
+public class IstioSetupService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IstioSetupService.class);
+
+    private static final String ISTIO_GATEWAY_NAMESPACE = "knative-eventing";
+    private static final String ISTIO_GATEWAY_NAME = "broker-gateway";
+    private static final String ISTIO_VIRTUAL_SERVICE_NAMESPACE = "knative-eventing";
+    private static final String ISTIO_VIRTUAL_SERVICE_NAME = "broker-virtual-service";
+    private static final String JWT_REQUEST_AUTHENTICATION_NAMESPACE = "istio-system";
+    private static final String JWT_REQUEST_AUTHENTICATION_NAME = "jwt-rh-sso";
+
+    @Inject
+    KubernetesClient kubernetesClient;
+
+    @Inject
+    TemplateProvider templateProvider;
+
+    @ConfigProperty(name = "event-bridge.istio.jwt.issuer")
+    String jwtIssuer;
+
+    @ConfigProperty(name = "event-bridge.istio.jwt.jwksUri")
+    String jwksUri;
+
+    void setupIstioComponent(@Observes StartupEvent event) {
+
+        createIstioGateway();
+
+        createIstioVirtualService();
+
+        createJWTRequestAuthentication();
+    }
+
+    private void createIstioGateway() {
+
+        Gateway existing = kubernetesClient.resources(Gateway.class)
+                .inNamespace(ISTIO_GATEWAY_NAMESPACE)
+                .withName(ISTIO_GATEWAY_NAME)
+                .get();
+
+        if (existing == null) {
+            Gateway expected = templateProvider.loadIstioGatewayTemplate();
+            expected.getMetadata().setName(ISTIO_GATEWAY_NAME);
+            expected.getMetadata().setNamespace(ISTIO_GATEWAY_NAMESPACE);
+
+            try {
+                kubernetesClient.resources(Gateway.class)
+                        .inNamespace(ISTIO_GATEWAY_NAMESPACE)
+                        .create(expected);
+            } catch (RuntimeException e) {
+                LOGGER.error(
+                        "Failed to create Istio Gateway resource. Please make sure it was properly deployed. The application keeps running due to https://issues.redhat.com/browse/MGDOBR-940 but the functionalitis are compromised.");
```

Comment:
We should be able to use some config to make this failure fatal in the "normal" scenario, in general the operator should stop in this case, it is just with CVP that we have a corner case.
@vaibhavjainwiz 
Wdyt?
I am thinking at some config param to pass to the operator

Reply:
There is no easy way to provide special configuration in CVP pipeline.

I am already in discussion with CVP team to check if it is possible to create a secret in CVP pipeline before start our operator. If it works then I could add this extra parameter in secret which would be mounted on our operator deployment.